research/ag_news/client_data.py

```diff
@@ -0,0 +1,80 @@
+from functools import partial
+from pathlib import Path
+from typing import Dict, List, Optional, Tuple
+
+import datasets
+import torch
+from datasets import load_dataset
+from torch.utils.data import DataLoader
+from transformers import BertTokenizer, PreTrainedTokenizer
+
+from fl4health.utils.dataset import DictionaryDataset
+from fl4health.utils.sampler import DirichletLabelBasedSampler
+
+
+def collate_fn_with_padding(
+    tokenizer: PreTrainedTokenizer, batch: List[Tuple[Dict[str, List[torch.Tensor]], torch.Tensor]]
+) -> Tuple[Dict[str, List[torch.Tensor]], torch.Tensor]:
+    """
+    Pad the sequences within a batch to the same length.
+    """
+    input_list, target_list = [], []
+    for pair in batch:
```

Comment:
Rather than indexing into pair here, it's a little more readable to unpack them in the for with something like 
```
for input_dict, target in batch:
....
```